server/src/main/java/com/linecorp/centraldogma/server/internal/mirror/OpenSSHPrivateKeyUtil.java

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright 2023 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.centraldogma.server.internal.mirror;
+
+import static net.i2p.crypto.eddsa.spec.EdDSANamedCurveTable.ED_25519_CURVE_SPEC;
+
+import java.math.BigInteger;
+import java.security.GeneralSecurityException;
+import java.security.KeyFactory;
+import java.security.KeyPair;
+import java.security.NoSuchProviderException;
+import java.security.PrivateKey;
+import java.security.PublicKey;
+import java.security.spec.ECParameterSpec;
+import java.security.spec.ECPoint;
+import java.security.spec.ECPrivateKeySpec;
+import java.security.spec.ECPublicKeySpec;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.RSAPrivateCrtKeySpec;
+import java.security.spec.RSAPublicKeySpec;
+import java.util.Arrays;
+import java.util.Objects;
+
+import org.apache.sshd.common.cipher.ECCurves;
+import org.apache.sshd.common.config.keys.KeyUtils;
+import org.apache.sshd.common.util.buffer.BufferUtils;
+import org.apache.sshd.common.util.security.SecurityUtils;
+import org.bouncycastle.util.Strings;
+import org.eclipse.jgit.util.Base64;
+
+import net.i2p.crypto.eddsa.EdDSAPrivateKey;
+import net.i2p.crypto.eddsa.EdDSAPublicKey;
+import net.i2p.crypto.eddsa.spec.EdDSAPrivateKeySpec;
+import net.i2p.crypto.eddsa.spec.EdDSAPublicKeySpec;
+
+final class OpenSSHPrivateKeyUtil {
+
+    static final byte[] BEGIN_OPENSSH = Strings.toByteArray("-----BEGIN OPENSSH PRIVATE KEY-----\n");
+    static final byte[] END_OPENSSH = Strings.toByteArray("\n-----END OPENSSH PRIVATE KEY-----");
+    static final byte[] AUTH_MAGIC = Strings.toByteArray("openssh-key-v1\0");
+
+    private static final String ED25519 = "ssh-ed25519";
+    private static final String ECDSA = "ecdsa";
+    private static final String RSA = "ssh-rsa";
+
+    static KeyPair parsePrivateKeyBlob(byte[] blob) {
+        // Referred to https://github.com/bcgit/bc-java/blob/43e0631b3c0ab5c0d660dc6bd72dd28633f3f839/core/src/main/java/org/bouncycastle/crypto/util/OpenSSHPrivateKeyUtil.java
```

Comment:
Then could you update our `NOTICE.txt` to mention that this project contains the code modified from Bouncy Castle?

Reply:
This class is removed completely.
Anyway, we haven't updated the `NOTICE.txt` file and the license folder. Let me make a separate PR for that. We should need a way to automate it by the way.